### ISSUE TITLE
toast.promise does not work with Jest

### ISSUE THREAD

**Issue by ashkan117** [emilkowalski/sonner#435]
### Describe the feature / bug 📝:

I'm currently using sonner like the following 
```tsx
toast.promise(apiPromiseCall, {
  loading: 'Loading...',
  success: (data) => {
    return `${data.name} toast has been added`;
  },
  error: 'Error',
});
```

When testing I want to mock out this api call but when used in tandem with sonner this fails with the following error.

```bash
  TypeError: n is not a function

    >  98 |     toast.promise(
          |           ^
       99 |       mockedApiCall(...),
      100 |       {
      101 |         loading: 'Loading message,
```

A workaround I am using is to manually handle the states with the `apiCall.then().catch()` methods

### Steps to reproduce the bug 🔁:

1. Mock a promise function using `jest.spyOn`
2. Trigger the event that calls the toast

I can try to add more details so let me know what else might be useful.

**Comment by ARKN-SOL** [comment]
I've stumbled upon this error in a different context. It's because sonner does an _outdated_ check if a promise is a `Promise`.

This is the responsible line of code: [src/state.ts#L124](https://github.com/emilkowalski/sonner/blob/bc74b2d62c7e0e7533a4a21c7597acefa837f231/src/state.ts#L124)
A promise doesn't need to be an instance of `Promise` to be a promise (yeah ... I know). It only needs to have a .then() method. This [Stackoverflow answer](https://stackoverflow.com/a/27746324) has a good summary.
TL;DR: Sonner should use `Promise.resolve(foo)` instead of testing for type.

**Comment by michalhonc** [comment]
Stumbled on this in the context of server actions.

Explicit one-off solution to what @ARKN-SOL suggested above

```diff
- toast.promise(apiPromiseCall, {
+ toast.promise(Promise.resolve(apiPromiseCall), {
  loading: 'Loading...',
  success: (data) => {
    return `${data.name} toast has been added`;
  },
  error: 'Error',
});
```

**Comment by emilkowalski** [comment]
Fixed in #565
